c++/src/kj/async-io-unix.c++

```diff
@@ -1472,7 +1472,7 @@ kj::Own<PeerIdentity> SocketAddress::getIdentity(kj::LowLevelAsyncIoProvider& ll
       // seen vague references on the internet saying that a PID of 0 and a UID of uid_t(-1) are used
       // as invalid values.
 
-#if defined(SO_PEERCRED)
+#if defined(SO_PEERCRED) && !__OpenBSD__
```

Comment:
I don't think so, this is intentional as OpenBSD requires `struct xucred creds` here. Reverting it will cause the following error:
```
src/kj/async-io-unix.c++:1476:20: error: variable has incomplete type 'struct ucred'
      struct ucred creds;
                   ^
src/kj/async-io-unix.c++:1476:14: note: forward declaration of 'ucred'
      struct ucred creds;
```

Reply:
It seems like I forget to push the fully reverted version last time, this file should be good now. Thanks for pointing that out.

Reply:
Can you add a comment here that OpenBSD defines `SO_PEERCRED` but uses a different interface for it, hence we're falling back to `LOCAL_PEERCRED`?

Reply:
Agree, done.